docs/deprecation.md

```diff
@@ -0,0 +1,39 @@
+# Deprecation Policy
+
+In order for any software to be reliable enough for production use, it needs to have a stable API. This means that
+users would have a high confidence that the way they use Kubevirt would not become obsolete, or at least would give
+the proper time to prepare if it does.
+
+This document aims to specify a process for several aspects of deprecation revolving Kubevirt. This is intended to
+be a "live document" which updates and refines itself over time.
+
+## Process
+
+### Feature Gates
+#### Introduction
+A feature gate should be introduced for every new feature, capability, configuration or any other change that
+is not yet stable and can jeopardize the cluster.
+
+A feature gate serves as:
+* A way to warn the user about a possible danger that could be caused by an unstable logic.
+* A way for the users to explicitly state that they would like to take the risk and enable the feature.
+
+Over time the new features would stabilize and graduate their API versions (i.e. from beta, to alpha to stable). When
+that happens, and therefore there risk of jeopardizing the user decreases significantly, the feature should be enabled
+by default and the feature gate should be deprecated (see section below).
+
+#### Deprecation
+When a feature stabilizes and there is no need to guard it by a feature gate, the feature gate should be deprecated.
+
+In order to deprecate a feature gate the following conditions must be met:
+* All the features that used to be guarded by it need to be enabled by default.
+  * The above point means that all of the features that were guarded by this feature gate would be enabled without
+  the need to enable this feature gate. In other words, the feature gate needs to turn into a no-op.
+* When the feature gate is enabled (by creating / updating Kubevit CR), a warning should be presented to the user. The
+  best fit to trigger the warning is probably our validating webhooks.
+
+We should keep the warning for the upcoming 4 releases <TODO: SHOULD BE FURTHER DISCUSSED>. Then, the feature gate
```

Comment:
Any opinions on this?

Reply:
i think the warning stays forever after the feature gate is deprecated... or at least we won't prevent the kubevirt cr from passing validation.